src/FACT_internal.c

```diff
@@ -1658,17 +1660,6 @@ void FACT_INTERNAL_OnBufferEnd(FAudioVoiceCallback *callback, void* pContext)
 		left
 	);
 
-	/* Last buffer in the stream? */
```

Comment:
Moved this till after we calculate the next offset, so we can reliably tell when we have exhausted the stream in the case that there is no looping of the wave going on. Using  audioBytes < stream "buffer" size is ambiguous when they end up being equal resulting in one extra unneeded callback.